core/cautils/parseFile.go

```diff
@@ -0,0 +1,217 @@
+				if isApiVersionEmpty == true || isKindEmpty == true {
+					err := fmt.Errorf("There is no enough objectID info")
+					return nil, err
+				}
+				splits := strings.Split(output, "dest")
+				if len(splits) < 2 {
+					err := fmt.Errorf("Something wrong with the length of the splits, which is %d", len(splits))
+					return nil, err
+				} else {
+					// cut the redundant one
+					splits = splits[1:]
+					for _, split := range splits {
+						path := extractParameter(pathRe, split, "$path")
+						pathType := extractParameter(typeRe, split, "$type")
+						mapMatched, err := regexp.MatchString(`!!map`, pathType)
+						if err != nil {
+							err = fmt.Errorf("regexp.MatchString err: %s", err.Error())
+						}
+						if mapMatched {
+							newlastNumber, err := writeNodeInfo(split, lastNumber, path, fileName, node, objectID, true)
+							lastNumber = newlastNumber
+							if err != nil {
+								err = fmt.Errorf("map type: writeNodeInfo wrong err: %s", err.Error())
+								return nil, err
+							}
+							mappingNodes.Nodes = append(mappingNodes.Nodes, *node)
+						} else {
+							newlastNumber, err := writeNodeInfo(split, lastNumber, path, fileName, node, objectID, false)
+							lastNumber = newlastNumber
+							if err != nil {
+								err = fmt.Errorf("not map type: writeNodeInfo wrong err: %s", err.Error())
+								return nil, err
+							}
+							mappingNodes.Nodes = append(mappingNodes.Nodes, *node)
+						}
+
+					}
+				}
+			}
+
+		}
+	}
+	return mappingNodes, nil
+}
+
+func New(MappingNodes MappingNodes) {
+	panic("unimplemented")
+}
+
+func writeNodeInfo(split string, lastNumber int, path string, fileName string, node *MappingNode, objectID *ObjectID, isMapType bool) (int, error) {
+	value, lineNumber, newLastNumber, err := getInfoFromOne(split, lastNumber, isMapType)
+	if err != nil {
+		err = fmt.Errorf("getInfoFromOne wrong err: %s", err.Error())
+		return -1, err
+	}
+	lastNumber = newLastNumber
+	node.writeInfoToNode(objectID, path, lineNumber, value, fileName)
+	return lastNumber, nil
+}
+
+func getInfoFromOne(output string, lastNumber int, isMapType bool) (value string, lineNumber int, newLastNumber int, err error) {
+	if isMapType == true {
+		value = ""
+	} else {
+		value = extractParameter(valueRe, output, "$value")
+	}
+	number := extractParameter(commentRe, output, "$line")
+	if number != "" {
+		lineNumber, err = strconv.Atoi(number)
+		if err != nil {
+			err = fmt.Errorf("strconv.Atoi err: %s", err.Error())
+			return "", -1, -1, err
+		}
+		if isMapType == true {
+			lineNumber = lineNumber - 1
+		}
+		lastNumber = lineNumber
+		// save to structure
+	} else {
+		lineNumber = lastNumber
+		// use the last one number
+	}
+	newLastNumber = lineNumber
+	return value, lineNumber, newLastNumber, nil
+}
+
+func getYamlLineInfo(index int, yamlFile string) (string, error) {
+	expression := `..| select(line == ` + strconv.Itoa(index) + `)| {"destpath": path | join("."),"type": type,"value": .}`
```

Comment:
Instead of constructing the expression on every function call, I would declare it outside the function:
```
var lineExpression = `..| select(line == %d)| {"destpath": path | join("."),"type": type,"value": .}`
```

and re-use it like this:
```
expression := fmt.Sprintf(lineExpression, index)
```

Reply:
Fix